### PR TITLE
state: cast file size to string

### LINKED ISSUE #1485: state: investigate excessive checksum re-calculations
From discord:

> I added the file, which computed checksum.  Then I dvc pushed, which computed checksum.  Then I dvc run -d thatfile, which computed checksum after the command finished.  Then I dvc pushed again (to make sure it wasn't going to push the dir that I untarred), which is computing the checksum again.

### PR DESCRIPTION
**Pull request by pared**
Fixes #1485

### CODE REVIEW

File: tests/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from mock import MagicMock
+
+
+def spy(method_to_decorate):
```
Comment (mroutis) [inline tests/utils/__init__.py]: :+1:

Comment (mroutis) [inline tests/utils/__init__.py]: :+1:

Comment (pared) [reply tests/utils/__init__.py]: Inspired by Spring and Stack Overflow

**Comment by pared** [comment]
@shcheklein sure, I'm on it.

**Comment by mroutis** [comment]
@shcheklein, don't hesitate to request changes with the GitHub interface :)

**Comment by pared** [comment]
@shcheklein how about now? :)

**Comment by shcheklein** [comment]
@pared this is great! thanks :) 👍

File: tests/unit/state.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from dvc.state import State
+from tests.basic_env import TestDvc
+
+
+class TestMtimeAndSizeReturnType(TestDvc):
+    def test(self):
+        self.create(self.DATA, self.DATA_CONTENTS)
```
Comment (efiop) [inline tests/unit/state.py]: Already created https://github.com/iterative/dvc/blob/0.23.2/tests/basic_env.py#L59 :slightly_smiling_face: 
```suggestion
        self.create(self.DATA, self.DATA_CONTENTS)
```

File: tests/unit/state.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        time, size = State.mtime_and_size(self.DATA)
```
Comment (efiop) [inline tests/unit/state.py]: Can we test the same for self.DATA_DIR too? Just so this unit test triggers the whole method.

**Pull request merged**